src/Pandell.PackageRunner/Program.cs

```diff
@@ -215,17 +248,49 @@ private int Run(Assembly packageRunnerAssembly, string packageRunnerExeFileName,
             using (var catalog = new AggregateCatalog(assemblies))
             using (var container = new CompositionContainer(catalog))
             {
+                writeVerboseMessage("Resolving execution package entry point.");
                 container.SatisfyImportsOnce(this);
                 if (this.RunAssembly == null)
                 {
-                    Console.Error.WriteLine("Method not found!");
+                    Program.WriteDiagnosticsMessage(TraceLevel.Error, "Execution package extry point not found!");
                     return 5;
                 }
-                this.RunAssembly(configuration.token);
+                writeVerboseMessage("Invoking execution package extry point.");
+                this.RunAssembly(configuration.token, Program.WriteDiagnosticsMessage);
+                writeVerboseMessage("Execution package finished successfully.");
                 return 0;
             }
         }
 
+        /// <summary>
+        /// </summary>
+        public static void WriteDiagnosticsMessage(TraceLevel traceLevel, string message)
```

Comment:
My idea was that the write function (used both by package runner and executable package) would be a closure that wraps around the requested level of verbosity. That way you simply keep invoking the trace message with level appropriate for each message and it internally checks whether that message is to be written out:

```c#
private static Func<TraceLevel, string, bool> CreateTraceWriter(TraceLevel maxLevel) {
    return (messageLevel, message) => {
        if (messageLevel > maxLevel) { return false; }
        if (message != null) {
            switch (messageLevel) { ... }
        }
        return true;
    }
}
```

The above would allow:

```c#
var defaultTraceLevel = TraceLevel.Warning;
var writer = CreateTraceWriter(defaultTraceLevel);

// complex message (do not want to execute if excluded)
if (writer(TraceLevel.Info, null)) { // check if trace enabled, no output
    writer(TraceLevel.Info, "Imported invoices:");
    foreach (var invoice in invoices) { writer(TraceLevel.Info, string.Format(...)); }
}

// use writer before command line arguments have been parsed
if (startupError) {
    writer(TraceLevel.Error, "Something went bad");
}

// change writer after command line arguments have been parsed
var requestedTraceLevel = ...;
if (requestedTraceLevel != defaultTraceLevel) {
    writer = CreateTraceWriter(requestedTraceLevel);
}
```